### ISSUE TITLE
Smarty errors on PHP8.1

### ISSUE THREAD

**Issue by verstratenbram** [modxcms/revolution#16299]
### Summary
When editing a modx resource on PHP8.1 the error log is flooded with PHP warnings.

A fix was implemented some time ago but this no longer seems to work: https://github.com/modxcms/revolution/commit/b48cd89b92e1c35a1b1a05ac5a6861bb336be257.

### Step to reproduce
Use PHP 8.1 and open a modx resource for editing.

### Observed behavior
The error log fills up with PHP warnings:

```
[2022-11-15 16:10:29] (ERROR @ /.../e81c1f54d865302a381a8f37098466e7d903ed88_0.file.image.tpl.php : 80) PHP warning: Undefined array key "wctx"
[2022-11-15 16:10:29] (ERROR @ /.../e81c1f54d865302a381a8f37098466e7d903ed88_0.file.image.tpl.php : 83) PHP warning: Undefined array key "openTo"
[2022-11-15 16:10:29] (ERROR @ /.../1e0e67492291b36bfe05f598ec6a7b15795f1f44_0.file.listbox-single.tpl.php : 62) PHP warning: Undefined array key "maxHeight"
[2022-11-15 16:10:29] (ERROR @ /.../1e0e67492291b36bfe05f598ec6a7b15795f1f44_0.file.listbox-single.tpl.php : 62) PHP warning: Undefined array key "maxHeight"
[2022-11-15 16:10:29] (ERROR @ /.../1e0e67492291b36bfe05f598ec6a7b15795f1f44_0.file.listbox-single.tpl.php : 62) PHP warning: Undefined array key "maxHeight"
[2022-11-15 16:10:29] (ERROR @ /.../bb6e37e1907b4512c38fee7198090617a6624d6a_0.file.listbox-multiple.tpl.php : 75) PHP warning: Undefined array key "maxHeight"
[2022-11-15 16:10:29] (ERROR @ /.../bb6e37e1907b4512c38fee7198090617a6624d6a_0.file.listbox-multiple.tpl.php : 75) PHP warning: Undefined array key "maxHeight"
[2022-11-15 16:10:29] (ERROR @ /.../1e0e67492291b36bfe05f598ec6a7b15795f1f44_0.file.listbox-single.tpl.php : 62) PHP warning: Undefined array key "maxHeight"
[2022-11-15 16:10:29] (ERROR @ /.../1e0e67492291b36bfe05f598ec6a7b15795f1f44_0.file.listbox-single.tpl.php : 62) PHP warning: Undefined array key "maxHeight"
[2022-11-15 16:10:29] (ERROR @ /.../1e0e67492291b36bfe05f598ec6a7b15795f1f44_0.file.listbox-single.tpl.php : 62) PHP warning: Undefined array key "maxHeight"
[2022-11-15 16:10:29] (ERROR @ /.../1e0e67492291b36bfe05f598ec6a7b15795f1f44_0.file.listbox-single.tpl.php : 62) PHP warning: Undefined array key "maxHeight"
[2022-11-15 16:10:29] (ERROR @ /.../1e0e67492291b36bfe05f598ec6a7b15795f1f44_0.file.listbox-single.tpl.php : 62) PHP warning: Undefined array key "maxHeight"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Undefined array key "showCheckbox"
[2022-11-15 16:10:29] (ERROR @ /.../10fb907bcacf8f9defa76f9752726f760bb8eea9_0.file.tvs.tpl.php : 64) PHP warning: Attempt to read property "value" on null
```

### Expected behavior
No errors in the logs.

### Environment
MODX version 2.8.3-pl, PHP 8.1

**Comment by JoshuaLuckers** [comment]
Thanks for taking time to report this issue and help make MODX better.

Two questions:
1. Would it be possible for you to test this using MODX 2.8.4?
2. Can you try and see if it helps when you clear the cache (remove the `smarty` directory in `core/cache/mgr`)?

**Comment by verstratenbram** [comment]
The same results happen when using MODX 2.8.4 and a cleared cache.

**Comment by verstratenbram** [comment]
I was able to suppress the warnings by adding 

```php
$this->muteExpectedErrors();
```

to the `modSmarty` [constructor](https://github.com/modxcms/revolution/blob/2.x/core/model/modx/smarty/modsmarty.class.php#L51).

I've created pull request #16301.

**Comment by Ibochkarev** [comment]
Fixed #16301
